[nokia] add new devices, update some `eol: false`

moviuro opened the pull request:
Data pulled from https://www.nokia.com/phones/en_int/security-updates where available

* https://www.nokia.com/phones/en_int/nokia-xr-21 was available for sale (not preorder) in France on 2023-05-24; will need to update EoL when more data becomes available on https://www.nokia.com/phones/en_int/security-updates
* C32 was released on 2023-02-25 according to https://www.gsmarena.com/nokia_c32-12139.php

moviuro committed:
[nokia] add new devices, update some `eol: false`

Data pulled from https://www.nokia.com/phones/en_int/security-updates where available

* https://www.nokia.com/phones/en_int/nokia-xr-21 was available for sale (not preorder) in France on 2023-05-24; will need to update EoL when more data becomes available on https://www.nokia.com/phones/en_int/security-updates
* C32 was released on 2023-02-25 according to https://www.gsmarena.com/nokia_c32-12139.php

marcwrobel commented on products/nokia.md:
```diff
@@ -11,13 +11,33 @@ activeSupportColumn: false
+    eol: 2025-12-01
+    link: https://www.nokia.com/phones/en_int/nokia-g-60
+-   releaseCycle: "Nokia X30 5G"
+    releaseDate: 2022-08-06
```
Shouldn't this be 2022-08-26 (according to https://www.nokia.com/phones/en_int/security-updates) ?

moviuro replied:
https://www.hmdglobal.com/nokia-x30-press-release says 2022-09-01, so I've picked that :wink:

marcwrobel reviewed (changes requested):
Hi @moviuro, just a small date to verify but the rest LGTM.

moviuro committed:
[nokia] fix release date for X30 5G

See https://www.hmdglobal.com/nokia-x30-press-release.
https://www.nokia.com/phones/en_int/security-updates mentions "Ramp up Production" on 2022-08-26, making an opinionated choice of picking the official press release date.

marcwrobel reviewed (approved):
LGTM, merging.
